mypy/newsemanal/semanal.py

```diff
@@ -2315,11 +2316,19 @@ def analyze_name_lvalue(self,
             added = self.add_symbol(name, var, lvalue)
             # Only bind expression if we successfully added name to symbol table.
             if added:
+                lvalue.is_new_def = True
+                lvalue.is_inferred_def = True
```

Comment:
FWIW, I just copied the logic from `make_name_lvalue_var()`. My understanding is that `process_type_annotation()` which is called _after_ `analyze_lvalues()` will set it to False (depending on annotation).

TBH, the logic is quite opaque here, and should be clean-up. I would however not do this because analyzers are merged, because it is very easy to break checker. I will create a follow-up issue.

Reply:
Opened https://github.com/python/mypy/issues/6458